plugins/node/opentelemetry-instrumentation-net/src/net.ts

```diff
@@ -0,0 +1,237 @@
+} from '@opentelemetry/instrumentation';
+import { GeneralAttribute } from '@opentelemetry/semantic-conventions';
+import { Net } from './types';
+import { VERSION } from './version';
+import { platform } from 'os';
+import { Socket } from 'net';
+
+const IPC_TRANSPORT = platform() == 'win32' ? 'pipe' : 'Unix';
+
+export class NetInstrumentation extends InstrumentationBase<Net> {
+  constructor(protected _config: InstrumentationConfig = {}) {
+    super('@opentelemetry/instrumentation-net', VERSION, _config);
+  }
+
+  init(): InstrumentationNodeModuleDefinition<Net>[] {
+    return [
+      new InstrumentationNodeModuleDefinition<Net>(
+        'net',
+        ['*'],
+        moduleExports => {
+          diag.debug('Applying patch for net module');
+          if (isWrapped(moduleExports.Socket.prototype.connect)) {
+            this._unwrap(moduleExports.Socket.prototype, 'connect');
+          }
+          this._wrap(
+            moduleExports.Socket.prototype,
+            'connect',
+            // eslint-disable-next-line @typescript-eslint/no-explicit-any
+            this._getPatchedConnect() as any
+          );
+          return moduleExports;
+        },
+        moduleExports => {
+          if (moduleExports === undefined) return;
+          diag.debug('Removing patch from net module');
+          this._unwrap(moduleExports.Socket.prototype, 'connect');
+        }
+      ),
+    ];
+  }
+
+  private _getPatchedConnect() {
+    return (original: (...args: unknown[]) => void) => {
+      const plugin = this;
+      return function patchedConnect(this: Socket, ...args: unknown[]) {
+        const options = normalizedArgs(args);
+
+        if (!options) {
+          startGenericSpan(plugin.tracer, this);
+          return original.apply(this, args);
+        }
+
+        const span = options.path
+          ? startIpcSpan(plugin.tracer, options, this)
+          : startTcpSpan(plugin.tracer, options, this);
+
+        return safeExecuteInTheMiddle(
+          () => original.apply(this, args),
+          error => {
+            if (error !== undefined) {
+              span.setStatus({
+                code: SpanStatusCode.ERROR,
+                message: error.message,
+              });
+              span.end();
+            }
+          }
+        );
+      };
+    };
+  }
+}
+
+interface NormalizedOptions {
+  host?: string;
+  port?: number;
+  path?: string;
+}
+
+function normalizedArgs(args: unknown[]): NormalizedOptions | null | undefined {
+  const opt = args[0];
+  if (!opt) {
+    return;
+  }
+
+  switch (typeof opt) {
+    case 'number':
+      return {
+        port: opt,
+        host: typeof args[1] === 'string' ? args[1] : 'localhost',
+      };
+    case 'object':
+      if (Array.isArray(opt)) {
+        return normalizedArgs(opt);
+      }
+      return opt;
+    case 'string':
+      return {
+        path: opt,
+      };
+    default:
+      return;
+  }
+}
+
+const SOCKET_EVENTS = ['connect', 'error', 'close'];
```

Comment:
please convert this to some enum / const so that it can be referenced instead of adding them directly later when for example you are removing events or listening to `connect` etc. etc.